src/components/Button/index.tsx

```diff
@@ -0,0 +1,65 @@
+import { Loading } from '@components/Loading'
+import { HTMLAttributes } from 'react'
+import styled, { css, DefaultTheme } from 'styled-components'
+
+export interface ButtonProps extends HTMLAttributes<HTMLButtonElement> {
+  children: string
+  size: 'large' | 'medium'
+  disabled?: boolean
+  isLoading?: boolean
+  block: boolean
+}
+
+const sizeModifiers = {
+  medium: (theme: DefaultTheme) => css`
+    height: ${theme.spacings.large};
+    font-size: ${theme.fonts.sizes.large};
+  `,
+  large: (theme: DefaultTheme) => css`
+    height: ${theme.spacings.xlarge};
+    font-size: ${theme.fonts.sizes.xlarge};
+  `,
+}
+
+const Wrapper = styled.button<Omit<ButtonProps, 'children'>>`
+  ${({ theme, size, disabled, block }) => css`
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    width: ${block ? '100%' : 'auto'};
+    padding: 0 1rem;
+    color: ${theme.colors.white};
+    background-color: ${theme.colors.black};
+    font-weight: bold;
+    border-radius: 0.2rem;
+    cursor: pointer;
+    ${size && sizeModifiers[size](theme)};
+    ${disabled &&
+    css`
+      opacity: 0.3;
+      pointer-events: none;
+    `}
+    :hover {
+      background-color: ${theme.colors.lightBlack};
+    }
+    :focus {
+      border: 0.2rem solid ${theme.colors.grey};
```

Comment:
Com borda funciona também, mas ele causa uma alteração no box model do botão. O ideal seria fazer com box-shadow, já que ele não altera o tamanho do botão, saca?
Dá uma olhada no que rola com focus/active:

https://user-images.githubusercontent.com/20327229/186277521-ffe59058-074a-479d-b67b-a96fbab07574.mov

Reply:
Fix in this commit [b3f936](https://github.com/joaooliveiradev/conduit-frontend/commit/b3f936b09ce78097e33e7b417758fdc109c34e7c)